source/framework/vx_node.c

```diff
@@ -1112,21 +1113,31 @@ static uint16_t ownNodeGetObjDescId(vx_node node, uint32_t pipeline_id)
 
 void ownNodeSetObjDescParamDirection(vx_node node)
 {
-    uint32_t prm_idx, prm_dir, is_prm_input;
+    uint32_t prm_idx, prm_dir, is_prm_input, is_prm_biput;
 
     is_prm_input = 0;
+    is_prm_biput = 0;
 
     for(prm_idx=0; prm_idx < ownNodeGetNumParameters(node); prm_idx++)
     {
         prm_dir = (uint32_t)ownNodeGetParameterDir(node, prm_idx);
 
         if(prm_dir==(uint32_t)VX_INPUT)
         {
-            tivxFlagBitSet(&is_prm_input, ((uint32_t)1<<prm_idx));
+            tivxFlagBitSet(&is_prm_input, ((uint32_t)1U<<prm_idx));
+        }
+        if(prm_dir==(uint32_t)VX_BIDIRECTIONAL)
```

Comment:
Do we need to check the prm_dir twice, as I see you are checking it again in line 1133?